enclave/trust_requirement.go

```diff
@@ -0,0 +1,279 @@
+package enclave
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	enclaveTrustRequirement "github.com/enclave-networks/go-enclaveapi/data/trustrequirement"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type trustRequirementResourceType struct{}
+
+func (t trustRequirementResourceType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:     types.Int64Type,
+				Computed: true,
+			},
+			"description": {
+				Type:     types.StringType,
+				Required: true,
+			},
+			"notes": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"user_authentication": {
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"authority": {
+						Type:     types.StringType,
+						Required: true,
+					},
+					"tenant_id": {
```

Comment:
How will this work for non-azure authorities? 

Could we arrange it like(?):

`
  user_authentication {
   azure {
    tenant_id: ""
   }
  }
`